### PR TITLE
add FindBoringSSL.cmake

### PR DESCRIPTION
**Pull request by wwieclaw**
# Testing
Tested in related PRs.

# JIra
https://swift-nav.atlassian.net/browse/BUILD-679

# Related PRs
* https://github.com/swift-nav/orion_proto/pull/1501
* https://github.com/swift-nav/orion/pull/5748

### CODE REVIEW

File: FindBoringSSL.cmake [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+# be be distributed together with this source. All other rights reserved.
+#
+# THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+# EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+#
+
+include("GenericFindDependency")
+GenericFindDependency(
+  TARGET ssl
```
Comment (isaactorz) [inline FindBoringSSL.cmake:15]: nit: I does boringssl create an alias target? i.e. `boringssl::ssl`

If so I would prefer we use those versions as `ssl` is really generic and cmake doesn't have any sort of namespacing.

File: FindBoringSSL.cmake [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  SOURCE_DIR boringssl
```
Comment (isaactorz) [inline FindBoringSSL.cmake:16]:
```suggestion
  ADDITIONAL_TARGETS crypto
  SOURCE_DIR boringssl
```

Lets add `ADDITIONAL_TARGETS` so we also make sure that the crypto library was found and configured correctly.

File: FindBoringSSL.cmake (hunk 1 above)
Comment (wwieclaw) [reply FindBoringSSL.cmake:15]: Yeah I was dissapointed to find out that it doesn't

**Review by isaactorz — APPROVED** [review]
lgtm!

**Pull request merged**